src/libsystemd/sd-event/sd-event.c

```diff
@@ -146,6 +156,12 @@ static int pending_prioq_compare(const void *a, const void *b) {
         if (x->enabled == SD_EVENT_OFF && y->enabled != SD_EVENT_OFF)
                 return 1;
 
+        /* Non rate-limited ones first. */
+        if (!x->ratelimited && y->ratelimited)
+                return -1;
+        if (x->ratelimited && !y->ratelimited)
+                return 1;
```

Comment:
```C
r = CMP(!x->ratelimited, !y->ratelimited);
if (r != 0)
    return r;
```
?

Reply:
Why the `!` though?

We want non-ratelimited ones first, i.e. the ones where the bool is 0, and that is the natural order of bools.

Reply:
hmm, without the !! it doesn't work though, as the CMP macro uses typeof() and that fails on a bit field. yuck.